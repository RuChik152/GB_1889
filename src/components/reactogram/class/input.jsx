import React, { PureComponent } from "react";

export class Input extends PureComponent {

    action = () => {
        this.props.click();
    }

    render() {
        return <div className = "text">
            <input className = "text__input" value = { this.props.text } onChange = { this.props.change } type = "text" placeholder = "Ваше сообщение" name = "msg" />
            <br />

            <button className = "text__btn" onClick = { this.action }>Отправить</button>
        </div>
    }

}