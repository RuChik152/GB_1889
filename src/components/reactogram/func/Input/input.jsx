import React from 'react';

export const Input = ({ text, change, click }) => {
  return (
    <div className="text">
      <input
        className="text__input"
        value={text}
        onChange={change}
        type="text"
        placeholder="Ваше сообщение"
        name="msg"
      />
      <br />
      <button className="text__btn" onClick={click} data-testid="test-id">
        Отправить
      </button>
    </div>
  );
};
