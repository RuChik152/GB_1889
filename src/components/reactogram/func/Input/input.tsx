import React, { FC, FormEvent, SetStateAction, useState} from 'react';
import style from './input.module.scss';
import { Button as ButtonUI, Input as InputUI } from '@mui/material';
import { useDispatch } from 'react-redux';
import { addMsg } from '../store/chats/action';
import { useParams } from 'react-router-dom';

export const Input: FC = () => {
  const [value, setValue] = useState('');
  const { chaiId } = useParams();
  const dispatch = useDispatch();

  const actionForm = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (chaiId) {
      dispatch(addMsg(chaiId, value));
    }
    setValue('');
  };
  return (
    <div className={style.text}>
      <form onSubmit={actionForm}>
        <InputUI
          id="text__input__id"
          value={value}
          onChange={(e) => setValue(e.target.value)}
          type="text"
          placeholder="Ваше сообщение"
        />
        <br />
        <ButtonUI
          disabled={!value}
          data-testid="test-id"
          variant="contained"
        >
          Нажмите Enter
        </ButtonUI>
      </form>
    </div>
  );
};
