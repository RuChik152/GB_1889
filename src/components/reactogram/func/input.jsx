import React, { useEffect, useRef } from 'react';

export const Input = (props) => {
  const inputEl = useRef(null);

  useEffect(() => {
    inputEl.current.focus();
  }, []);

  const action = () => {
    props.click();
  };

  return (
    <div className="text">
      <input
        className="text__input"
        value={props.text}
        onChange={props.change}
        type="text"
        placeholder="Ваше сообщение"
        name="msg"
        ref={inputEl}
      />
      <br />
      <button className="text__btn" onClick={action}>
        Отправить
      </button>
    </div>
  );
};
