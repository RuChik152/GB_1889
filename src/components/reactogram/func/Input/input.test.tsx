import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Input } from './input';
import userEvent from '@testing-library/user-event';

const user = userEvent.setup();

describe('Input', () => {
  it('Render', () => {
    render(<Input />);
  });

  it('is there an INPUT and BUTTON', () => {
    render(<Input />);
    expect(screen.getByPlaceholderText('Ваше сообщение')).toBeInTheDocument();
    expect(screen.getByTestId('test-id')).toBeInTheDocument();
  });

  it('input value to string', () => {
    render(<Input />);
    expect(screen.getByPlaceholderText('Ваше сообщение')).toHaveValue('text');
  });

  it('button 3 click', async () => {
    const func = jest.fn();
    render(<Input />);
    user.tripleClick(screen.getByTestId('test-id'));
    await waitFor(() => expect(func).toHaveBeenCalledTimes(3), {
      timeout: 1100,
    });
  });


  it('Snapshot', () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
});
