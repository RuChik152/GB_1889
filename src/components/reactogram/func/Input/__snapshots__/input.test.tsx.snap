// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot 1`] = `
<DocumentFragment>
  <div
    class="text"
  >
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-a88p61-MuiInputBase-root-MuiInput-root"
    >
      <input
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id="text__input__id"
        name="msg"
        placeholder="Ваше сообщение"
        type="text"
        value="text"
      />
    </div>
    <br />
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="test-id"
      tabindex="0"
      type="button"
    >
      Отправить
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
